src/Attachment/File.php

```diff
@@ -77,7 +84,9 @@ public function load(): Model
     {
         $attachment = $this->getMatchesHash();
 
-        if (! $this->storage->has($this->engine->path())) {
+        if (!is_null($this->path)) {
```

Comment:
I don't think testing in this way is a good solution. It would be much better if we changed this to allow passing a parameter to the engine class. An example that fits here and below:
```diff
- $this->path ?? $this->engine->path()
+ $this->engine->path($path)
```

And the engine class itself decided how to process this information—returning the same path.